src/IO/WriteBufferFromS3MemoryStream.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "config.h"
+
+#if USE_AWS_S3
+
+#include <iostream>
+
+namespace DB
+{
+
+struct MemoryStream: std::iostream
```

Comment:
`std::iostream` is a class, so I suggest making MemoryStream a class too. And usually we use public inheritance.